File: backend/PropertyAgency.Domain/Entities/Property.cs
using System.ComponentModel.DataAnnotations.Schema;
using PropertyAgency.Domain.Enums;

namespace PropertyAgency.Domain.Entities;

public class Property
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public int RoomCount { get; set; }
    public string Status { get; set; }
    
    public Guid AddressId { get; set; }
    public Address Address { get; set; }

    public List<Favorite> Favorites { get; set; }
    public List<Rental> Rentals { get; set; }

    public Property()
    {
        Favorites = new List<Favorite>();
        Rentals = new List<Rental>();
    }
    
}

File: frontend/assets/types/property.ts
export interface PropertyAddress {
    id: string;
    street: string;
    city: string;
    state: string;
    country: string;
    zipCode: string;
    propertyId: string;
}

export interface Favorites {
    id: string;
    userId: string;
    user: string;
    propertyId: string;
}

export interface Property {
    id: string;
    title: string;
    description: string;
    price: number;
    roomCount: number;
    status: string;
    images: string[];
    userId: string;
    user: {
        id: string;
        userName: string;
        email: string;
        role: number;
        properties: any[];
        favorites: Favorites[],
        rentals: any[]
    }
    address: PropertyAddress;
    favorites: Favorites[];
    rentals: any[];
}
